spring-cloud-dataflow-rest-resource/src/main/java/org/springframework/cloud/dataflow/rest/resource/DeploymentStatus.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.dataflow.rest.resource;
+
+import org.springframework.util.Assert;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+
+/**
+ * Provides a typed enumeration of deployment statuses for Streams.
+ *
+ * @author Gunnar Hillert
+ *
+ */
+@JsonFormat(shape = JsonFormat.Shape.OBJECT)
+public enum DeploymentStatus {
+
+	DEPLOYING("deploying", "Deploying" ,"The app or group is being deployed."),
+	DEPLOYED("deployed", "Deployed", "All apps have been successfully deployed"),
+	UNDEPLOYED("undeployed", "Undeployed", "The app or group is known to the system, but is not currently deployed"),
+	PARTIAL("partial", "Partial", "In the case of multiple apps, some have successfully deployed, while others have not"),
+	FAILED("failed", "Failed", "All apps have failed deployment"),
+	ERROR("error", "Error", "A system error occurred trying to determine deployment status"),
+	UNKNOWN("unknown", "Unknown", "The app or group deployment is not known to the system");
+
+	private final String key;
```

Comment:
Personally, I believe using the enum's name for **complex enumerations** is wrong. I ran into to many cases in my life where that bit me.

Also, that way, Jackson can nicely render the enum using:

```
@JsonFormat(shape = JsonFormat.Shape.OBJECT)
```